Controllers/HomeController.cs

```diff
@@ -3,21 +3,88 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
+using System.Text.RegularExpressions;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Newtonsoft.Json;
 
 namespace Zoltu.Bags.Client.Controllers
 {
 	public class HomeController : Controller
 	{
+		private static Regex productRegx = new Regex(@"product\/(\d+)", RegexOptions.IgnoreCase);
```

Comment:
Style: We could probably improve the readability of the code later on as well as reduce the chance for future bugs by using named captures.
```
var regex = new Regex(@"product\/(?<productId>\d+)");
var productId = regex.Match(path).Groups["productId"].Value.TryParseUInt64();

```
The `TryParseUInt64` extension method would need to be created (see other comment about this).

The same thing could be applied to all of the regular expressions that have match groups, though `tags` just returns a string so that wouldn't get run through the `TryParseUInt64` bit.